lib/cooked_post_processor.rb

```diff
@@ -384,7 +384,7 @@ def optimize_image!(img, upload, cropped: false)
     w, h = img["width"].to_i, img["height"].to_i
 
     # note: optimize_urls cooks the src and data-small-upload further after this
-    thumbnail = upload.thumbnail(w, h)
+    thumbnail = !upload.animated && upload.thumbnail(w, h)
```

Comment:
Maybe we should instead fix `thumbnail` to handle animated GIFs?

Reply:
`thumbnail` is sometimes good as it is now. For example [here](https://github.com/discourse/discourse/blob/6d4711ca109ffe2da4c7eb833bcbe32d481656db/lib/cooked_post_processor.rb#L362), it generates the "lazy load" thumbnail (the very low res, 10x10 image).